tls/s2n_record_read_aead.c

```diff
@@ -63,6 +64,7 @@ int s2n_record_parse_aead(
         GUARD(s2n_stuffer_write_bytes(&iv_stuffer, four_zeroes, 4));
         GUARD(s2n_stuffer_write_bytes(&iv_stuffer, sequence_number, S2N_TLS_SEQUENCE_NUM_LEN));
         for (int i = 0; i < cipher_suite->record_alg->cipher->io.aead.fixed_iv_size; i++) {
+	    S2N_INVARIENT(i <= cipher_suite->record_alg->cipher->io.aead.fixed_iv_size);
```

Comment:
should this be a patch to this file?

whats the criteria for applying a patch vs modifying the source?

Reply:
Patches are a little more fragile than inline annotations, so if it makes sense within the flow of the C code, adding an invariant inline is probably better.

For annotations that are specific to a particular proof, doing a patch makes more sense.